### ISSUE TITLE
Question: Does the code format differ between versions of python 3 (3.7, 3.8, 3.9) ?

### ISSUE THREAD

**Issue by freddyaboulton** [psf/black#2383]
Hello,

Not sure where the best place to ask this question is so I decided to file an issue. In my organization, we've been running black for a couple of months under python versions 3.7 and 3.8 in our CI and I've never noticed that the output is different between the two.

Is there any scenario where the code style between the python 3 versions will be different? I'm wondering if I should suggest that we only run black with python 3.8.

Thank you! I love using your package.

**Comment by HassanAbouelela** [comment]
Off the top of my head, you will get some differences when it comes to version-specific features, but only in very limited cases. For a minimal reproducible example, have a look at the following script:
```py
if x := False:
    print("what?")
```

Walrus was introduced 3.8 I believe, which means a 37 vs 38 black will have slightly different behavior (only in certain cases).
The following commands will work (file.py is the code above):
> (38) black file.py
> (38) black -t py37 file.py
> (38) black -t py37 -c "if x := False: print("wot")"

But not the following:
> (37) black file.py
> (37) black -t py38 file.py
> (37) black -t py38 -c "if x := False: print("wot")"

In short: yes, there is some difference in behavior (mostly in AST parsing in this case), but you may not notice it depending on your code. If your codebase is not intended to support 3.7, then it's probably safe to ignore this. If 3.7 is supported though, it may make sense to keep it.

**Comment by freddyaboulton** [comment]
Thank you @HassanAbouelela ! This is a very helpful answer.

**Comment by cooperlees** [comment]
I think the tl;dr here is just run black under the newest cpython runtime possible (or in use at your company) and you should have less issues.
